src/utilities/filetypes/AlfalfaJSON.cpp

```diff
@@ -0,0 +1,223 @@
+#include "AlfalfaJSON.hpp"
+#include "AlfalfaJSON_Impl.hpp"
+#include "utilities/alfalfa/AlfalfaActuator.hpp"
+#include "utilities/alfalfa/AlfalfaOutputVariable.hpp"
+#include "utilities/alfalfa/AlfalfaGlobalVariable.hpp"
+#include "utilities/alfalfa/AlfalfaMeter.hpp"
+#include "utilities/alfalfa/AlfalfaConstant.hpp"
+#include "../core/PathHelpers.hpp"
+#include "utilities/idd/IddObject.hpp"
+#include "utilities/idd/IddEnums.hpp"
+
+#include <utilities/idd/OS_EnergyManagementSystem_OutputVariable_FieldEnums.hxx>
+#include <utilities/idd/OS_EnergyManagementSystem_GlobalVariable_FieldEnums.hxx>
+#include <utilities/idd/OS_EnergyManagementSystem_Actuator_FieldEnums.hxx>
+#include <utilities/idd/OS_Output_Variable_FieldEnums.hxx>
+#include <utilities/idd/OS_Output_Meter_FieldEnums.hxx>
+#include <utilities/idd/EnergyManagementSystem_OutputVariable_FieldEnums.hxx>
+#include <utilities/idd/EnergyManagementSystem_GlobalVariable_FieldEnums.hxx>
+#include <utilities/idd/EnergyManagementSystem_Actuator_FieldEnums.hxx>
+#include <utilities/idd/Output_Variable_FieldEnums.hxx>
+#include <utilities/idd/Output_Meter_FieldEnums.hxx>
+#include <utilities/idd/IddEnums.hxx>
+
+#include <iostream>
+namespace openstudio {
+namespace alfalfa {
+  namespace detail {
+    AlfalfaJSON_Impl::AlfalfaJSON_Impl() {}
+
+    AlfalfaJSON_Impl::AlfalfaJSON_Impl(const std::string& s) {}
+
+    AlfalfaJSON_Impl::AlfalfaJSON_Impl(const openstudio::path& p) {}
```

Comment:
that **cannot** be right. You're not using the string/path.

Reply:
Yeah, this is a current debate I'm having on whether or not you should be able to create this object from JSON. It is not required by OpenStudio, but it might be nice to have for use of the file outside the workflow.

Either way I wanted to get the rest of the API figured out before writing a load of parsers.